### PR TITLE
greenfoot 3.9.0

### PR DESCRIPTION
**Pull request by eleanordoesntcode**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

---

I modified the install URL, since the ARM and Intel versions are now separated.

### CODE REVIEW

File: Casks/g/greenfoot.rb [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+  on_arm do
+    sha256 "ba33e14149e58ed95423b108e4b6c401c5dd822b36460ccee2ae71dcbc996261"
+
+    url "https://www.greenfoot.org/download/files/Greenfoot-mac-aarch64-#{version.no_dots}.dmg"
+  end
+  on_intel do
+    sha256 "ab6488f1d736248193fcc1e93f56e2b12c051b8b6888df55e2b9885f902b7071"
+
+    url "https://www.greenfoot.org/download/files/Greenfoot-mac-x64-#{version.no_dots}.dmg"
+  end
```
Comment (krehel) [inline Casks/g/greenfoot.rb:13]: This should use the `arch` stanza, not separate `on_` blocks.

Comment (eleanordoesntcode) [reply Casks/g/greenfoot.rb:13]: ah, didn't know it was possible, I'll modify it accordingly

**Comment by khipp** [comment]
@eleanordoesntcode You can fix style-related issues automatically by running `brew style --fix greenfoot.rb`.

**Comment by khipp** [comment]
Thank you! Could you please squash the commits for a cleaner history?

**Comment by eleanordoesntcode** [comment]
Done

**Review by khipp — APPROVED** [review]

**Pull request merged**